xdsl/frontend/block.py

```diff
@@ -1,7 +1,8 @@
 import ast
+from typing import Any
 
 
-def block(*params):
+def block(*params: Any):
```

Comment:
There were a number of type errors in the frontend files, @georgemitenkov, could you please take a look and lower that number? Do you have the squiggly lines when developing locally?

Reply:
No... and I should! Which linter/type checker do you use? Pyright?

Reply:
I use VSCode, with Pylance (which I think uses pyright under the hood)
```
"python.analysis.typeCheckingMode": "strict"
```

Reply:
```
(venv) (base) sasha@SL-MBP-16 ~/D/x/xdsl> pyright | grep -E 'xdsl/xdsl/frontend' | wc -l                                                                                                                                   main
stubPath /Users/sasha/Develop/xdslproject/xdsl/typings is not a valid directory.
      78
(venv) (base) sasha@SL-MBP-16 ~/D/x/xdsl> pyright | grep -E 'xdsl/xdsl/' | wc -l                                                                                                                                           main
stubPath /Users/sasha/Develop/xdslproject/xdsl/typings is not a valid directory.
     149
```